components/frontend-react/src/components/UserSignup.js

```diff
@@ -42,19 +42,53 @@ const useStyles = makeStyles(theme => ({
 
 const UserSignup = (props) => {
 
+  const [username, setUsername] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setconfirmPassword] = useState('');
+
+  const handleTextFields = (event) => {
+    switch(event.target.id)
+    {
+      case 'name':
+        setUsername(event.target.value);
+        return;
+
+      case 'email':
+        setEmail(event.target.value);
+        return;
+
+      case 'pass':
+        setPassword(event.target.value);
+        return;
+
+      case 'c_pass':
+        setconfirmPassword(event.target.value);
+        return;
+
+      default:
+        return;
+    }
+  }
+
   const handleSubmitButton = (event) => {
     event.preventDefault();
-    props.onClickSubmit();
+    let details = {user_name: username, email: email, password: password, confirm_password: confirmPassword};
+    if(password !== confirmPassword)
+    {
+      alert('Passwords do not match');
```

Comment:
I hope this is temporarily, later you will be replacing this with a model. 

I think it is better to add `TODO` comments,

```
// TODO Replace with a model dialog
alert('Passwords do not match');
```

Reply:
I thought of using react-toast component. I will add in my next PR

Reply:
Cool